### PR TITLE
Emc post ssmis f17

### LINKED ISSUE #366: Add SSMIS-F17 channels to grib2 output
Add SSMIS-F17 channels (channel 15, 16, 17, 18) to grib2 output for HAFS satellite products. There are parm file changes for hafs  in EMC_post/parm and source code changes (RQSTFLD.F, CALRAD_WCLOUD_newcrtm.f)

### PR DESCRIPTION
**Pull request by LinZhu-NOAA**
Add the SSIMS-F17 channels to grib2 output for HAFS development. The key modifications are as follows:
1. control files in parm directory 
a) organized non-satellite products and add 8 satellite products including 4 goes-r and 4 ssmis-f17 channels
postcntrl_hafs.xml 
postxconfig-NT-hafs.txt

b) add ssmis-f17 channels 
post_avblflds.xml 
arams_grib2_tbl_new

2. source code changes:
a) RQSTFLD.F
Line 2367-2379: add ssmis-f17 channels (825 - 828)

b) CALRAD_WCLOUD_newcrtm.f
Line 208: add local logical variable post_ssmis17
Line 290-294: set post_ssmis17 
Line 334: add post_ssmis17 to the list
Line 456-469: set ssmis17 channels (channel 15-18)
Line 485-487: take out the ssmis_f17 channel selection for grib1
Line 542: add ssmis_f17 
Line 1249: add ssmis_f17
Line 1808-1825: modify output block for ssmis_f17

For hafs regression tests, it need to link the crtm fix files to run directory since satellite products are added to grib2 control file. 
export FIXcrtm=/gpfs/dell1/nco/ops/nwprod/lib/crtm/v2.3.0/fix
/gpfs/hps3/emc/hwrf/noscrub/Lin.L.Zhu/save/hafs_20210527/ush/link_crtm_fix.sh ${FIXcrtm}

### CODE REVIEW

File: parm/params_grib2_tbl_new_ssmisf17 [hunk 1]
```diff
@@ -0,0 +1,1273 @@
+   0   7    11   0 4LFTX
```
Comment (WenMeng-NOAA) [inline parm/params_grib2_tbl_new_ssmisf17]: @LinZhu-NOAA Please do not start a new file. Any new variables should be added in params_grib2_tbl_new.txt and params_grib2_tbl_new (sorted table).

File: parm/postcntrl_hafs.xml [hunk 2]
```diff
@@ -92,16 +92,8 @@
...
-       <pname>TKE</pname>
-       <level>200. 500. 700. 1000. 2000. 3000. 5000. 7000. 10000. 12500. 15000. 17500. 20000. 22500. 25000. 27500. 30000. 32500. 35000. 37500. 40000. 42500. 45000. 
-47500. 50000. 52500. 55000. 57500. 60000. 62500. 65000. 67500. 70000. 72500. 75000. 77500. 80000. 82500. 85000. 87500. 90000. 92500. 95000. 97500. 100000.</level>
-       <scale>3.0</scale>
-       </param>
-
-       <param>
-       <shortname>CLMR_ON_ISOBARIC_SFC</shortname>
-       <pname>CLMR</pname>
+       <shortname>CLWMR_ON_ISOBARIC_SFC</shortname>
```
Comment (WenMeng-NOAA) [inline parm/postcntrl_hafs.xml:95]: @LinZhu-NOAA Can you list the variables removed and added in hafs products in the issue #366?

Comment (LinZhu-NOAA) [reply parm/postcntrl_hafs.xml:95]: Sure. I will list them in the issue #366 .

File: parm/params_grib2_tbl_new_ssmisf17 (hunk 1 above)
Comment (LinZhu-NOAA) [reply parm/params_grib2_tbl_new_ssmisf17]: For params_grib2_tbl_new, I compared the one that Bio sent to me and the one that I checked out from the EMC_post develop branch, there are several differences, not only ssmis-f17 variables. That's why I added a new params_grib2_tbl_new_ssmisf17 . Should I just add the ssmis-f17 variables to the params_grib2_tbl_new that checked out from the EMC_post develop branch and commit it?

Comment (LinZhu-NOAA) [reply parm/params_grib2_tbl_new_ssmisf17]: The params_grib2_tbl_new_ssmisf17 is removed and ssmis-f17 variables are added to params_grib2_tbl_new and params_grib2_tbl_new.text.

File: parm/postcntrl_hafs.xml (hunk 2 above)
Comment (LinZhu-NOAA) [reply parm/postcntrl_hafs.xml:95]: The variables added in and removed from hafs products are listed in the issue #366.

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA Can you also sync your branch with upstream/develop with the instructions at
https://github.com/NOAA-EMC/EMC_post/wiki/UPP-Code-Development#update-your-branch-at-your-fork-with-branch-develop-at-authoritative-repository ?

**Comment by LinZhu-NOAA** [comment]
@WenMeng-NOAA The featured branch ems_post_ssmis-f17 is synced with upstream/develop.

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA Thanks for the update.

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA To generate new dataset HURSAT,  the fix files from crtm module are needed. Can you provide me the configuration of your test so I can update the baseline of hafs in the UPP rt tests?

**Comment by LinZhu-NOAA** [comment]
@WenMeng-NOAA I ran the tests on Luna. I modified the scripts for HAFS regression tests as follows:

/gpfs/hps3/emc/hwrf/noscrub/Lin.L.Zhu/save/EMC_post_RT/jobs-dev/run_post_fv3hafs_WCOSS_C.sh
/gpfs/hps3/emc/hwrf/noscrub/Lin.L.Zhu/save/EMC_post_RT/jobs-dev/run_post_fv3hafs_pe_test_WCOSS_C.sh

eg. run_post_fv3hafs_WCOSS_C.sh
I set a FIXcrtm in line 65 and  link the crtm fix files to run directory in line 91.

Thanks,

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA I conducted the UPP standalone tests for hafs. Can you verify the grib2 data sets HURPRS09.tm00 and HURSAT09.tm00 at /u/Wen.Meng/ptmp/fv3hafs_2020082512 on mars? Thanks!

**Comment by LinZhu-NOAA** [comment]
@WenMeng-NOAA There is only HURPRS09.tm00 in /u/Wen.Meng/ptmp/fv3hafs_2020082512/ on Mars. I also check the /u/Wen.Meng/ptmp/fv3hafs_2020082512/postxconfig-NT.txt, there is no satellite products in this configure file, it seems old hafs configure file. Thanks!

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA Can you double check the directory /u/Wen.Meng/ptmp/fv3hafs_2020082512 from WCOSS-Dell Mars, not luna? The actual path is /gpfs/dell2/ptmp/Wen.Meng/fv3hafs_2020082512. Thanks!

**Comment by LinZhu-NOAA** [comment]
@WenMeng-NOAA Sorry, I checked the files in wrong directory. Yes, these output looks good to me. Thanks!

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA Thanks for confirming. I will proceed the final testing for merging. The UPP regression test baseline for hafs will be recreated with this PR merged.

**Comment by LinZhu-NOAA** [comment]
@WenMeng-NOAA Thanks a lot! BTW, will you generate the new lib for in-line post after merging? Thanks!

**Comment by WenMeng-NOAA** [comment]
@LinZhu-NOAA The new upp lib will be generated in September. The NCEPLIBs team would like the libs at the hpc-stack not upgraded too frequently.

**Comment by LinZhu-NOAA** [comment]
@WenMeng-NOAA Okay, got it. Thank you very much!

File: sorc/ncep_post.fd/CALRAD_WCLOUD_newcrtm.f [hunk 3]
```diff
@@ -205,6 +205,7 @@ SUBROUTINE CALRAD_WCLOUD
   logical post_abig16, post_abig17, post_abigr ! if true, user requested at least one abi channel
   logical fix_abig16, fix_abig17   ! if true, abi_g16, abi_g17 fix files are available
   logical post_ahi8 ! if true, user requested at least on ahi channel (himawari8)
+  logical post_ssmis17 ! if true, user requested at least on ssmis_f17 channel
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/CALRAD_WCLOUD_newcrtm.f:209]: @LinZhu-NOAA Can you add a change log with your name next to the line 14 in this routine? Thanks!

File: sorc/ncep_post.fd/RQSTFLD.F [hunk 4]
```diff
@@ -2364,9 +2364,18 @@ module RQSTFLD_mod
      &                      /1,'F16 SSMIS NON-NADIR ',118,109,     &
      &                       'SSMIS TB top_of_atmos '/ !table 133
 
-!    SSMIS_F17 (L(1)-L(7) -> ID8: 201-207 -> 183H, 19H, 19V, 37H, 37V, 85H, 85V)
+!    SSMIS_F17 (SSMIS_F17 (CH15:37H, CH16:37V, CH17:91H, CH18:91V))
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/RQSTFLD.F:2368]: @LinZhu-NOAA Can you also add a change log in this routine? Thanks!

File: sorc/ncep_post.fd/CALRAD_WCLOUD_newcrtm.f (hunk 3 above)
Comment (LinZhu-NOAA) [reply sorc/ncep_post.fd/CALRAD_WCLOUD_newcrtm.f:209]: @WenMeng-NOAA The change log is added, please check. Thanks!

File: sorc/ncep_post.fd/RQSTFLD.F (hunk 4 above)
Comment (LinZhu-NOAA) [reply sorc/ncep_post.fd/RQSTFLD.F:2368]: @WenMeng-NOAA The change log is added in this subroutine. Please check. Thanks!

**Comment by WenMeng-NOAA** [comment]
The UPP regression tests were conducted on WCOSS-Dell, WCOSS-Cray and Hera. With this PR merged, the baseline of hafs will be re-created. This PR is ready for merging.

**Review by WenMeng-NOAA — APPROVED** [review]

**Pull request merged**